.github/workflows/scorecards.yml

```
@@ -0,0 +1,73 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
+
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '36 2 * * 5'
+  push:
+    branches: [ "master" ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

```suggestion
        uses: checkout@v4
```

Consistent with other workflow

---

IMO I think we should not change this as in the scorecard action there is a check for [pinned dependencies](https://github.com/ossf/scorecard/blob/cfe5c5f9c3dbf2a98c864e1b1ea25fae289f3b9d/docs/checks.md#pinned-dependencies) (this project has many workflows which are not pinned by hash resulting score of 0 which needs improvement) so if we change this it will further drop the score. And as for updates Dependabot updates the pinned hash when needed.

---

get it. Pinning actions to a full-length commit SHA is a recommended practice. We could push for the overhaul of other workflows following the merge of this PR. I noticed that the latest version of `checkout` is `v4.17`, using `actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7` directly could reduce the need for updates.

---

> And as for updates Dependabot updates the pinned hash when needed.

Do you mean using a pinned dependencies will not block Dependabot, right?
I asked the same question at https://github.com/karmada-io/karmada/pull/5396#discussion_r1721117214.

---

Yes, I think it will not block it.

---

Thanks for the clarification. LGTM then.
/assign @zhzhuang-zju